migrate_test.go

```diff
@@ -56,3 +62,153 @@ func validateMigrationSort(t *testing.T, ms Migrations, sorted []int64) {
 
 	t.Log(ms)
 }
+
+func TestCollectMigrations(t *testing.T) {
+	// Not safe to run in parallel
+	t.Run("no_migration_files_found", func(t *testing.T) {
+		tmp := t.TempDir()
+		err := os.MkdirAll(filepath.Join(tmp, "migrations-test"), 0755)
+		check.NoError(t, err)
+		_, err = collectMigrationsFS(os.DirFS(tmp), "migrations-test", 0, math.MaxInt64, nil)
+		check.HasError(t, err)
+		check.Contains(t, err.Error(), "no migration files found")
+	})
+	t.Run("filesystem_registered_with_single_dirpath", func(t *testing.T) {
+		t.Cleanup(func() { clearMap(registeredGoMigrations) })
+		file1, file2 := "09081_a.go", "09082_b.go"
+		file3, file4 := "19081_a.go", "19082_b.go"
+		AddNamedMigrationContext(file1, nil, nil)
+		AddNamedMigrationContext(file2, nil, nil)
+		check.Number(t, len(registeredGoMigrations), 2)
+		tmp := t.TempDir()
+		dir := filepath.Join(tmp, "migrations", "dir1")
+		err := os.MkdirAll(dir, 0755)
+		check.NoError(t, err)
+		createEmptyFile(t, dir, file1)
+		createEmptyFile(t, dir, file2)
+		createEmptyFile(t, dir, file3)
+		createEmptyFile(t, dir, file4)
+		fsys := os.DirFS(tmp)
+		files, err := fs.ReadDir(fsys, "migrations/dir1")
+		check.NoError(t, err)
+		check.Number(t, len(files), 4)
+		all, err := collectMigrationsFS(fsys, "migrations/dir1", 0, math.MaxInt64, registeredGoMigrations)
+		check.NoError(t, err)
+		check.Number(t, len(all), 4)
+		check.Number(t, all[0].Version, 9081)
+		check.Number(t, all[1].Version, 9082)
+		check.Number(t, all[2].Version, 19081)
+		check.Number(t, all[3].Version, 19082)
+	})
+	t.Run("filesystem_registered_with_multiple_dirpath", func(t *testing.T) {
```

Comment:
@rafalneves this test aims to capture what _I think_ you're trying to do, but if you could look over this specific subtest that'd be great.